db/crud.go

```diff
@@ -311,6 +315,10 @@ func (db *Database) GetDelta(docID, fromRevID, toRevID string) (delta *RevisionD
 				return nil, &redactedBody, nil
 			}
 
+			if fromRevision.IsRemoval() {
```

Comment:
As discussed, this check should happen immediately after revision cache retrieval.  (delta can't be generated if we don't have the fromRevision body)

Reply:
Sure, this has been moved to right after revision cache retrieval.